File: max_sottointervallo/turingarena-files/Java/skeleton.java
import java.util.Scanner;

abstract class Skeleton {
    private static final Scanner in = new Scanner(System.in);


    abstract int max_sottointervallo(int N, int seq[]);

    public static void main(String[] args) {
        Solution __solution = new Solution();

        // checkpoint
        System.out.printf("%d\n", 0);
        // read N
        int N;
        System.out.flush();
        N = in.nextInt();
        // for i to N {...}
        int[] seq;
        seq = new int[N];
        for (int i = 0; i < N; i++) {
            // read seq[i]
            System.out.flush();
            seq[i] = in.nextInt();
        }
        // call index = max_sottointervallo(N, seq)
        int index;
        index = __solution.max_sottointervallo(N, seq);
        // write index
        System.out.printf("%d\n", index);
        // exit
        System.exit(0);
    }
}

File: max_sottointervallo/turingarena-files/Rust/skeleton.rs
mod solution;
use std::io::Write;

macro_rules! readln {
    ($($var:expr),*) => {{
        let mut buf = String::new();
        std::io::stdin().read_line(&mut buf).unwrap();
        let parts: Vec<&str> = buf.trim().split(" ").collect();
        let mut i: usize = 0;
        $(
            assert!(i < parts.len(), "input format incorrect: too few values on this line"); 
            $var = parts[i].parse().unwrap();
            i += 1;
        )*
        assert!(i == parts.len(), "input format incorrect: too many values on the line")
    }};
}


fn main() {
    // checkpoint
    println!("{}", 0);
    // read N
    let N: i64;
    std::io::stdout().flush().unwrap();
    readln!(N);
    // for i to N {...}
    let mut seq: Vec<i64> = Vec::new();
    seq.resize(N as usize, 0);
    for i in 0..N {
        // read seq[i]
        std::io::stdout().flush().unwrap();
        readln!(seq[i as usize]);
    }
    // call index = max_sottointervallo(N, seq)
    let index: i64;
    index = solution::max_sottointervallo(N, seq);
    // write index
    println!("{}", index);
    // exit
    std::process::exit(0);
}
